packages/circuit-ui/components/NotificationInline/NotificationInline.tsx

```diff
@@ -0,0 +1,267 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HTMLAttributes, RefObject, useEffect, useRef, useState } from 'react';
+import { css } from '@emotion/react';
+import { Alert, Confirm, Info, Notify } from '@sumup/icons';
+
+import styled, { StyleProps } from '../../styles/styled';
+import { useAnimation } from '../../hooks/useAnimation';
+import Body from '../Body';
+import { TrackingProps } from '../../hooks/useClickEvent';
+import CloseButton from '../CloseButton';
+import { hideVisually } from '../../styles/style-mixins';
+import Button, { ButtonProps } from '../Button';
+import { ClickEvent } from '../../types/events';
+
+const TRANSITION_DURATION = 200;
+const DEFAULT_HEIGHT = 'auto';
+
+type Variant = 'info' | 'confirm' | 'notify' | 'alert';
+
+type Action = ButtonProps;
+
+type CloseProps =
+  | {
+      /**
+       * Renders a close button in the top right corner and calls the provided function
+       * when the button is clicked.
+       */
+      onClose: (event: ClickEvent) => void;
+      /**
+       * Text label for the close button for screen readers.
+       * Important for accessibility.
+       */
+      closeButtonLabel: string;
+    }
+  | { onClose?: never; closeButtonLabel?: never };
+
+export type BaseProps = HTMLAttributes<HTMLDivElement> & {
+  /**
+   * Notification inline variants. Defaults to `info`.
+   */
+  variant?: Variant;
+  /**
+   * Notification inline headline to provide information (optional)
+   */
+  headline?: string;
+  /**
+   * A body copy to provide information
+   */
+  body: string;
+  /**
+   * An optional call-to-action button.
+   */
+  action?: Action;
+  /**
+   * Whether the notification inline is visible.
+   */
+  isVisible?: boolean;
+  /**
+   * Additional data that is dispatched with the tracking event.
+   */
+  tracking?: TrackingProps;
+  /**
+   * A clear and concise description of the icon and the Toast's purpose. If the toast body is self-explanatory pass an empty string.
+   */
+  iconLabel?: string;
+};
+
+export type NotificationInlineProps = BaseProps & CloseProps;
+
+// TODO: update the design token colors to be info/confirm/alert/notify, then remove this mapping
+const colorMap = {
+  info: 'p500',
+  confirm: 'success',
+  alert: 'danger',
+  notify: 'warning',
+} as const;
+
+const iconMap = {
+  info: Info,
+  confirm: Confirm,
+  alert: Alert,
+  notify: Notify,
+};
+
+const inlineWrapperStyles = () => css`
+  display: inline-flex;
```

Comment:
In the end I think that this doesn't work: I played around with the stories to create a more "real-life" scenario, where the notification is on top of a block of text (could also be a card or something):

<img width="1678" alt="Screenshot 2022-01-17 at 11 35 28" src="https://user-images.githubusercontent.com/35560568/149754093-1c617e95-5112-43f1-ac59-27882a817a15.png">

With `inline` (or `inline-flex`), it keeps the width of the content and there would be no way (as far as I can see) to make it stretch without overriding the component's CSS.

Instead we probably want something like this in most cases:

<img width="1678" alt="Screenshot 2022-01-17 at 11 37 27" src="https://user-images.githubusercontent.com/35560568/149754267-19045d43-2607-4e9e-9979-0af7456051c5.png">

So I suggest we remove this line. If developers still need to make the notification fit the content, they can also do it by adding a wrapper, which I think is cleaner than overriding styles.

Reply:
_Side note:_

Also, my related suggestion in #1312 is probably not the best choice: just like the `NotificationInline`, it should stretch in some cases according to the specs

<img width="757" alt="Screenshot 2022-01-17 at 11 42 04" src="https://user-images.githubusercontent.com/35560568/149755069-eae08b7e-1b76-45a3-a1c1-2cc0abd0411d.png">

I'll check with design what the expected visual is when it stretches _and_ has an asset (because that's the scenario when it looks unbalanced right now). I'll post an update in the issue at #1312

Reply:
So you suggest to leave it as is? And for banner too, with being stretched in some cases. And leave to the teams to adjust if necessary?

Reply:
Yes for the inline notification 🙂 as I mentioned, it's going to be easier to constrain the width than to stretch a component that's constrained by default, so I think it's a better solution. What do you think @amelako/@connor-baer?

For the banner I reached out to design like I said and we'll see what they suggest, it's a similar situation with the added challenge of having an optional asset that can make the banner look unbalanced. But let's take that discussion to #1312